packages/block-editor/src/components/navigable-toolbar/index.js

```diff
@@ -1,32 +1,21 @@
 /**
  * External dependencies
  */
-import { cond, matchesProperty, omit } from 'lodash';
+import { omit } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { NavigableMenu, KeyboardShortcuts } from '@wordpress/components';
 import { Component, createRef } from '@wordpress/element';
 import { focus } from '@wordpress/dom';
-import { ESCAPE } from '@wordpress/keycodes';
-
-/**
- * Browser dependencies
- */
-
-const { Node, getSelection } = window;
 
 class NavigableToolbar extends Component {
 	constructor() {
 		super( ...arguments );
 
 		this.focusToolbar = this.focusToolbar.bind( this );
-		this.focusSelection = this.focusSelection.bind( this );
 
-		this.switchOnKeyDown = cond( [
-			[ matchesProperty( [ 'keyCode' ], ESCAPE ), this.focusSelection ],
```

Comment:
This changes / regresses a behavior:

1. Create paragraph
2. Press Alt+F10
3. Press Escape

**With top toolbar:**

Before: Escape placed caret back into paragraph
After: Nothing happens when Escape is pressed

**Without top toolbar:**

Before: Escape placed caret back into paragraph
After: Block navigation mode is triggered

Reply:
It was an intended change. We discussed this with some folks (@ellatrix and someone else I don't remember :P). The idea being that with this navigation mode, the "Escape" to move back to the block doesn't make sense anymore.

Now for the differences between "top" and "docked" toolbar. I'd argue that that's reasonable enough (even if not perfect).

 - Anything by the block triggers navigation mode,
 - Anything outside the block (inspector, top toolbar) doesn't.

Reply:
In the case of the toolbar attached to the block, you can just tab back to the place you were. Not sure about the top toolbar though. There tabbing moves to through everything in the editor. `Esc` returns to the block, but in navigation mode. Maybe that's fine?